README.md

```diff
@@ -36,9 +36,35 @@ date. After cloning the repo:
 $ git submodule init && git submodule update
 ```
 
+### testing
+
+This crate uses a number of testing strategies, although all the tests can be
+run with `cargo test`.
+
+In addition to unit tests, we have a custom property-testing system for testing
+parsing and compilation. This involves data that is stored in the `test-data`
+directory. These tests work by taking some input, producing some output, and
+then comparing that output to an expected output that is included in the test
+data. There are three different formats for this:
+
+- *parse tests*: These ensure that we generate the expected AST or errors for
+  various inputs. See [`test-data/parse-tests/README.md`][parse readme] for an
+  overview.
+- *compile tests*: These ensure that we generate the expected [TTX output][ttx]
+  or errors for a various inputs. See [`test-data/compile-tests/README.md`][compile readme] for an overview.
+- *fonttools tests*: Very similar to the compile tests, but instead of being cases
+  that we define ourselves, we reuse the property tests from [fonttools
+  feaLib][feaLib tests]. This ensures that we generate equivalent output to
+  feaLib.
+
 ## architecture sketch
 
-The overall design of this crate is heavily inspired by the design of [rust analyzer].
+The overall design of this crate is heavily inspired by the design of [rust
+analyzer]. At a very high level, given some source, we:
+- *lex* the source file into raw tokens
+- *parse* these tokens into an abstract syntax tree
+- *validate* this tree, ensuring that it conforms with the specification
+- *compile* the validated tree into OpenType tables.
```

Comment:
Do you literally mean each of those steps is 100% separate? We don't lex and parse and maybe even mix in some validation in the same pass? - fairly common to muddle up those phases. Can we link to the source that does the job from here?